aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -48,7 +48,7 @@ void window_function_checks(
       " is not implemented for sparse types, got: ",
       options);
   AT_CHECK(
-      at::isFloatingType(options.dtype()),
+      at::isFloatingType(dataTypeToScalarType(options.dtype().id())),
```

Comment:
Are you planning to follow up on this one? Fixing this particular case properly does not seem too difficult: we have to basically have a new notion of "PyTorch natively recognized floating point dtype" (the point being that, in this test, we don't actually care if a type is floating point or not, but that it is one of the floating point types we have actually compiled support for.)